C/common/reading_set.cpp

```diff
@@ -192,6 +192,45 @@ ReadingSet::append(const vector<Reading *>& readings)
 	}
 }
 
+/**
+* Deep copy a set of readings to this reading set.
+*/
+bool
+ReadingSet::copy(const ReadingSet& src)
+{
+   vector<Reading *> *readings = new vector<Reading *>;
+   bool copyResult = true;
+   try
+   {
+	   for ( auto rs : src.getAllReadings())
+	   {
+		   std::string assetName = rs->getAssetName();
+		   for ( auto dp : rs->getReadingData())
+		   {
+			   std::string dataPointName  = dp->getName();
+			   DatapointValue dv = dp->getData();
+			   Datapoint *value = new Datapoint(dataPointName, dv);
+			   Reading *in = new Reading(assetName, value);
+			   readings->push_back(in);
+		   }
+	   }
+   }
+   catch(...)
+   {
+	   copyResult = false;
+	   readings->clear();
+	   Logger::getLogger()->error("Copy opeation failed");
```

Comment:
It woudl be good to have a little more inforamation. Imagine you are looking at a problem reported by a customer and you see "Copy operation failed" in the log - would it help you find the problem? You do not not what object its s trying to copy? Did it fail because it ran out of memory or was there a problem wit the thing it was copying?

It is always wise to look at you error messages and imagine if it would mean anything to a random engineer 12 months from now.

Reply:
There is a typo in that error message: opeation -> operation.